server/webapp/WEB-INF/rails.new/spec/controllers/api_v1/admin/pluggable_scms_controller_spec.rb

```diff
@@ -0,0 +1,310 @@
+      end
+
+      it 'should not allow rename of material name' do
+        controller.stub(:check_for_stale_request).and_return(nil)
+        params = {material_name: 'foo', id: '1', name: 'material', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+
+        put_with_api_header :update, params
+
+        expect(response.code).to eq('406')
+        expect(actual_response[:message]).to eq('Renaming of SCM material is not supported by this API.')
+
+      end
+
+      it 'should not update existing material if validations fail' do
+        controller.stub(:check_for_stale_request).and_return(nil)
+        controller.stub(:check_for_scm_rename).and_return(nil)
+        result = double('HttpLocalizedOperationResult')
+        HttpLocalizedOperationResult.stub(:new).and_return(result)
+        result.stub(:isSuccessful).and_return(false)
+        result.stub(:message).with(anything()).and_return("Update failed")
+        result.stub(:httpCode).and_return(422)
+        @pluggable_scm_service.should_receive(:updatePluggableScmMaterial).with(anything, an_instance_of(SCM), result)
+        params = {material_name: 'material'}
+
+        put_with_api_header :update, params
+
+        expect(response.code).to eq('422')
+        expect(actual_response[:message]).to eq("Update failed")
+      end
+
+      it 'should fail update if etag does not match' do
+        controller.request.env['HTTP_IF_MATCH'] = "some-etag"
+        params = {material_name: 'foo', id: '1', name: 'foo', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+        @pluggable_scm_service.should_receive(:findPluggableScmMaterial).with('foo').and_return(@scm)
+
+        put_with_api_header :update, params
+
+        expect(response.code).to eq("412")
+        expect(actual_response).to eq({:message => "Someone has modified the global SCM 'foo'. Please update your copy of the config with the changes."})
+      end
+
+      it 'should proceed with update if etag matches the etag ' do
+        hash_for_existing_scm = ApiV1::Scms::PluggableScmRepresenter.new(@scm).to_hash(url_builder: controller)
+        controller.request.env['HTTP_IF_MATCH'] = "\"#{Digest::MD5.hexdigest(JSON.generate(hash_for_existing_scm))}\""
+        params = {material_name: 'material', id: '1', name: 'material', auto_update: false, plugin_configuration: {id: 'github.pr', version: '1'}, configuration: [{"key" => 'url', "value" => 'git@github.com:foo/bar.git'}]}
+
+        @pluggable_scm_service.should_receive(:findPluggableScmMaterial).with('material').and_return(@scm)
+        @pluggable_scm_service.should_receive(:updatePluggableScmMaterial).with(anything, an_instance_of(SCM), anything)
+
+        put_with_api_header :update, params
+
+        expect(response.code).to eq("200")
```

Comment:
Since the response body is going to be a contract, perhaps a test for the body is better?